internal/controller/todo.go

```diff
@@ -0,0 +1,99 @@
+package controller
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/daniyalumer/todo-list-go-chi/internal/helper"
+	"github.com/daniyalumer/todo-list-go-chi/internal/models"
+	"github.com/daniyalumer/todo-list-go-chi/internal/service"
+	"github.com/go-chi/chi/v5"
+)
+
+func TodoRoutes() chi.Router {
+	r := chi.NewRouter()
+
+	r.Get("/", GetTodosHandler)
+	r.Post("/", CreateTodoHandler)
+	r.Put("/", UpdateTodoHandler)
+	r.Delete("/", DeleteTodoHandler)
+
+	return r
+}
+
+func GetTodosHandler(w http.ResponseWriter, r *http.Request) {
+	todos, err := service.ReadTodoList()
+	if err != nil {
+		helper.TodoListResponseWriter(w, "failed to read todo list", []models.Todo{}, http.StatusBadRequest)
```

Comment:
In case of failure, we only return err messages. In case of success, we will either return the response object or success message.